homeassistant/components/evse_wifi/sensor.py

```diff
@@ -0,0 +1,487 @@
+"""Platform for EVSE Sensor integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    UnitOfElectricCurrent,
+    UnitOfEnergy,
+    UnitOfLength,
+    UnitOfPower,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add sensors for passed config_entry in HA."""
+    evse = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = EvseCoordinator(hass, evse)
+
+    new_devices = [
+        VehicleStateSensor(coordinator, evse),
+        VehicleStateTextSensor(coordinator, evse),
+    ]
+    new_devices.append(MaxCurrentSensor(coordinator, evse))
+    new_devices.append(ActualPowerSensor(coordinator, evse))
+    new_devices.append(DurationSensor(coordinator, evse))
+    new_devices.append(LastActionUserSensor(coordinator, evse))
+    new_devices.append(LastActionUidSensor(coordinator, evse))
+    new_devices.append(EnergySensor(coordinator, evse))
+    new_devices.append(MilageSensor(coordinator, evse))
+    new_devices.append(MeterReadingSensor(coordinator, evse))
+    new_devices.append(CurrentP1Sensor(coordinator, evse))
+    new_devices.append(CurrentP2Sensor(coordinator, evse))
+    new_devices.append(CurrentP3Sensor(coordinator, evse))
```

Comment:
Instead of creating unique sensor objects, use a sensorentitydescription. Look in the code base for examples.